src/bigquery.ts

```diff
@@ -805,12 +874,12 @@ export class BigQuery extends common.Service {
    * @returns {object} A properly-formed `queryParameter` object.
    */
   // tslint:disable-next-line no-any
-  static valueToQueryParameter_(value: any) {
+  static valueToQueryParameter_(value: any, providedType: any = undefined) {
     if (is.date(value)) {
       value = BigQuery.timestamp(value as Date);
     }
 
-    const parameterType = BigQuery.getType_(value);
+    const parameterType = BigQuery.getType_(value, providedType);
```

Comment:
Could we do an if/else for calling the "getProvidedType_" path when `providedType` is defined, and the "getType_" when it isn't?

Reply:
Done.